protocol/src/BridgeDepositWithdraw.sol

```diff
@@ -137,23 +157,67 @@ contract BridgeDepositWithdraw is Bridge, IBridgeDepositWithdraw {
     /**
      * @inheritdoc IBridgeDepositWithdraw
      */
-    function claimNative(PodECDSA.CertifiedLog calldata certifiedLog) public override whenNotPaused {
-        (bytes32 id, uint256 amount, address to) = _decodeDepositNativeLog(certifiedLog.log);
-
+    function claimNative(uint256 amount, AttestedTx.AttestedTx calldata attested, bytes calldata aggregated_signatures, MerkleTree.MultiProof calldata proof) public override whenNotPaused {
         if (!_isValidTokenAmount(MOCK_ADDRESS_FOR_NATIVE_DEPOSIT, amount, false)) revert InvalidTokenAmount();
 
-        bytes32 requestId = _hashRequest(id, address(0), amount, to);
-        if (processedRequests[requestId]) revert RequestAlreadyProcessed();
+        console.log("claimNative called with amount: %s", amount);
+        console.log("Merkle proof path:");
+        for (uint i = 0; i < proof.path.length; i++) {
+            console.logBytes32(proof.path[i]);
+        }
+        console.log("Merkle proof flags:");
+        for (uint i = 0; i < proof.flags.length; i++) {
+            console.logBool(proof.flags[i]);
+        }
 
-        if (certifiedLog.log.addr != bridgeContract) revert InvalidBridgeContract();
+        // Build leaves for validating merkle proof
+        bytes4 selector = bytes4(keccak256("depositNative()"));
+        bytes32[] memory leaves = new bytes32[](4);
+        leaves[0] = MerkleTree.hashLeaf("from", keccak256(abi.encode(msg.sender)));
+        leaves[1] = MerkleTree.hashLeaf("to", keccak256(abi.encode(bridgeContract)));
+        leaves[2] = MerkleTree.hashLeaf("value", keccak256(abi.encode(amount)));
+        leaves[3] = MerkleTree.hashLeaf("input", keccak256(abi.encodePacked(selector)));
 
-        bool verified = PodECDSA.verifyCertifiedLog(podConfig, certifiedLog);
-        if (!verified) revert InvalidCertificate();
+        sortLeaves(leaves);
```

Comment:
we are doing a lot of unnecessary work here. we shouldn't use the merkle tree as a set commitment but rather as a fixed size vector commitment. just fix the positions of the leaves (e.g. the first leaf is sender, then contract address, ...) and then you just need to check the exact index. also would avoid hashing the key as the position of the leave would already give you that.